decidim-consultations/app/permissions/decidim/consultations/admin/permissions.rb

```diff
@@ -0,0 +1,107 @@
+# frozen_string_literal: true
+
+module Decidim
+  module Consultations
+    module Admin
+      class Permissions < Decidim::DefaultPermissions
+        def permissions
+          return permission_action unless user
+          return permission_action unless permission_action.scope == :admin
+
+          if !user.admin?
+            disallow!
+            return permission_action
+          end
+
+          user_can_enter_space_area?
+
+          if read_admin_dashboard_action?
+            allow!
+            return permission_action
+          end
+
+          allowed_read_participatory_space?
+          allowed_consultation_action?
+          allowed_question_action?
+          allowed_response_action?
+
+          permission_action
+        end
+
+        private
+
+        def question
+          @question ||= context.fetch(:question, nil)
+        end
+
+        def consultation
+          @consultation ||= context.fetch(:consultation, nil) || context.fetch(:participatory_space, nil)
+        end
+
+        def response
+          @response ||= context.fetch(:response, nil)
+        end
+
+        def allowed_consultation_action?
+          return unless permission_action.subject == :consultation
+
+          case permission_action.action
+          when :create, :read, :publish
+            allow!
+          when :update, :destroy, :preview
+            toggle_allow(consultation.present?)
```

Comment:
Sorry to ask (I know this API was probably introduced in another PR) but what does `toggle_allow` do, exactly? It sounds a bit confusing / misleading.

Reply:
Can be changed in #3137 if needed. If the condition evaluates to `true`, it calls `allow!` on the permission, otherwise it calls `disallow!` on it.